### ISSUE TITLE
Introduce Enriched mesh_seq/solver object

### ISSUE THREAD

**Issue by ddundo** [mesh-adaptation/goalie#261]
To reduce bloat of `GoalOrientedMeshSeq` (which will be made even worse by #259), it would be nice to introduce an `EnrichedMeshSeq` object (or `EnrichedSolver` if #238 goes through). And ideally we'd be able to interact with it for testing purposes etc. - at the moment it's not accessible outside of `indicate_errors`.

**Comment by jwallwork23** [comment]
I agree although would prefer the `EnrichedSolver` route. We should make a separation of concerns between `MeshSeq` and `Solver`, making `MeshSeq` handle just the meshes.
